pkg/operator/storageclasscontroller/storageclasscontroller_test.go

```diff
@@ -187,3 +191,53 @@ func TestSync(t *testing.T) {
 		})
 	}
 }
+
+func TestSyncMultiple(t *testing.T) {
+	tests := []struct {
+		name                   string
+		storagePolicySyncFails bool
+		expectError            bool
+	}{
+		{
+			name:                   "when policy sync is successful",
+			storagePolicySyncFails: false,
+			expectError:            false,
+		},
+		{
+			name:                   "when policy sync failed",
+			storagePolicySyncFails: true,
+			expectError:            true,
+		},
+	}
+
+	for i := range tests {
+		test := tests[i]
+		t.Run(test.name, func(t *testing.T) {
+			initialObjects := []runtime.Object{testlib.GetConfigMap(), testlib.GetSecret()}
+			clusterCSIDriverObject := testlib.MakeFakeDriverInstance()
+			configObjects := runtime.Object(testlib.GetInfraObject())
+			commonApiClient := testlib.NewFakeClients(initialObjects, clusterCSIDriverObject, configObjects)
+			storageClass := "storageclass1.yaml"
+			apiDeps := getCheckAPIDependency(commonApiClient)
+			var conn *vclib.VSphereConnection
+			scController := newStorageClassController(commonApiClient, storageClass, test.storagePolicySyncFails)
+			scController.backoff = defaultBackoff
+
+			// err will be nil on even on failure, need to check conditions instead
+			err := scController.Sync(context.TODO(), conn, apiDeps)
+			if err != nil && !test.expectError {
+				t.Errorf("failed to sync controller: %+v", err)
+			}
```

Comment:
Same question here, it's a little confusing to see a comment "err will be nil on failure" followed by checking `err != nil`.

Reply:
I have rewritten the actual test code to be better at catching error conditions.